server/graphql/v2/mutation/AccountMutations.ts

```diff
@@ -0,0 +1,37 @@
+import { GraphQLNonNull } from 'graphql';
+import GraphQLJSON from 'graphql-type-json';
+import { AccountReferenceInput, fetchAccountWithReference } from '../input/AccountReferenceInput';
+import { Account } from '../interface/Account';
+import { Unauthorized, Forbidden } from '../../errors';
+
+const accountMutations = {
+  editAccountSettings: {
+    type: new GraphQLNonNull(Account),
+    description: 'Edit the settings for the given account',
+    args: {
+      account: {
+        type: new GraphQLNonNull(AccountReferenceInput),
+        description: 'Account where the settings will be updated',
+      },
+      settings: {
+        type: new GraphQLNonNull(GraphQLJSON),
+        description: 'Settings to set for this account',
+      },
+    },
+    async resolve(_, args, req): Promise<object> {
+      if (!req.remoteUser) {
+        throw new Unauthorized();
+      }
+
+      const account = await fetchAccountWithReference(args.account, { loaders: req.loaders, throwIfMissing: true });
+
+      if (!req.remoteUser.isAdmin(account.id)) {
+        throw new Forbidden();
+      }
+
+      return account.update({ settings: args.settings });
```

Comment:
Should we consider any blacklist/whitelist strategy?

Should we consider merging by default? To not erase existing settings?

Reply:
A work has already been started to implement whitelisting at the collective level: https://github.com/opencollective/opencollective-api/blob/a0400e1f4b18cf259a0f22c9f9449c463f5f242c/server/models/Collective.js#L317. We need to improve that because it's not a hard whitelist right now, but it's a different issue: we must ensure we don't break any existing settings. A problem induced by whitelisting is that any change will then require an API update, it's handy to be able to update the settings without touching the API.

Regarding the merge vs erase, while I agree it's safer, the major issue is that it will prevent keys from being removed.

Reply:
The new `editAccountSetting` whitelists the `key` being submitted.